geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/MembershipBuilderImpl.java

```diff
@@ -99,9 +108,12 @@ public MembershipBuilderImpl(
   public Membership<ID> create() throws MembershipConfigurationException {
     GMSMembership<ID> gmsMembership =
         new GMSMembership<>(membershipListener, messageListener, lifecycleListener);
-    Services<ID> services =
+    final Services<ID> services =
         new Services<>(gmsMembership.getGMSManager(), statistics, authenticator,
             membershipConfig, serializer, memberFactory, locatorClient, socketCreator);
+    if (membershipLocator != null) {
+      services.setLocators(membershipLocator.getLocator(), membershipLocator);
```

Comment:
important addition here: if a membership locator was specified, then tell the services about it

Reply:
when running in a regular member that is not a locator